release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,238 @@
+            }
+            enum BOTH {
+                description "Use both the IPv6 source and
+                    destination addresses in the calculation of the
+                    hash.";
+            }
+            enum TARGET_DEFINED {
+                description "Use the default fields that are used by
+                    the device for calculating the hash.";
+            }
+            enum FLOW_LABEL {
+                description "Use the flow label in the IPv6 header
+                    to calculate the hash.";
+            }
+        }
+        description "The IPv6 fields that should be used to
+            compute the hash.";
+    }
+
+    typedef hash-field-type-transport {
+        type enumeration {
+            enum NONE {
+                description "Do not use any transport fields in the
+                    calculation of the hash.";
+            }
+            enum SRC_PORT {
+                description "Use only the source port from the
+                    transport header in the calculation of the hash.";
+            }
+            enum DST_PORT {
+                description "Use only the destination port from the
+                    transport header in the calculation of the hash.";
+            }
+            enum BOTH {
+                description "Use both the source and
+                    destination ports from the transport header in
+                    the calculation of the hash.";
+            }
+            enum TARGET_DEFINED {
+                description "Use the default fields that are used by
+                    the device for calculating the hash.";
+            }
+        }
+        description "The trasport layer fields that should be used to
+            compute the hash.";
+    }
+
+    grouping hashing-inputs {
+        description
+            "Top level container for inputs to be used for hashing policies.";
+
+        leaf ipv4 {
+            type hash-field-type-ipv4;
+            description "The IPv4 fields that should be used to
+                compute the hash.";
+        }
+        leaf ipv6 {
+            type hash-field-type-ipv6;
+            description "The IPv6 fields that should be used to
+                compute the hash.";
+        }
+        leaf transport-ports {
+            type hash-field-type-transport;
+            description "The transport layer fields that should be used to
+                compute the hash.";
+        }
+    }
+
+    grouping hashing-policy-config {
+        description
+          "Configuration data for hashing policies.";
+
+        leaf name {
+            type string;
+            description
+              "The name of the hashing policy.
+
+              When a configured user-controlled policy is created by the
+              system, it is instantiated with the same name in the
+              /policies/policy[name]/state list.";
+        }
+
+        leaf algorithm {
+            type string;
```

Comment:
This seems extremely open as a definition. What does this string mean? What are the possible values, and how do we validate them? It should at the very least be an `identity` that has a common set of values that are widely implemented, and then a subsequent set of potential extensions that can be added to the same base identity if there are hardware/vendor specific options that are needing to be represented here.

Reply:
Proposed solution in #959 requiring the implementation to publish a list of algorithms that are implementation defined.  Please comment on #959